### ISSUE TITLE
Gradient update error

### ISSUE THREAD

**Issue by dstx123** [OpenHelix-robot/OpenHelix#2]
For the embedding layer, this method does not seem to update the gradient properly. p is a torch.nn.Parameter. Its requires_grad attribute can only be set for the entire parameter tensor, not for a part of the tensor.

```
    for n, p in model.named_parameters():
        if any(
            [
                x in n
                for x in ["text_hidden_fcs","pred_act_mlps","pred_pos_act","pred_rot_act", "pred_gripper_act"]
            ]
        ):
            print("n: ", n, "p.shape: ", p.shape)
            p.requires_grad = True
        if "embed_tokens.weight" in n:
            p[32003:].requiers_grad = True
```

**Comment by dstx123** [comment]
Thank you for your answer and this excellent work. @OpenHelix-robot
